artemis-cli/src/main/java/org/apache/activemq/artemis/cli/commands/QueueStats.java

```diff
@@ -0,0 +1,195 @@
+      this.queueName = queueName;
+   }
+
+   public void setExactMatch(boolean exactMatch) {
+      this.exactMatch = exactMatch;
+   }
+
+   public void setMaxRows(int maxRows) {
+      this.maxRows = maxRows;
+   }
+
+   public void setUser(String user) {
+      this.user = user;
+   }
+
+   public void setPassword(String password) {
+      this.password = password;
+   }
+
+   @Override
+   public Object execute(ActionContext context) throws Exception {
+      super.execute(context);
+      String filter = createFilter(queueName, exactMatch);
+
+      if (verbose) {
+         context.out.println("queueName='" + queueName + "'");
+         context.out.println("maxRows='" + maxRows + "'");
+         context.out.println("exactMatch='" + exactMatch + "'");
+         context.out.println("filter='" + filter + "'");
+      }
+
+      printStats(context, filter, maxRows);
+      return null;
+   }
+
+   private void printStats(final ActionContext context, final String filter, int maxRows) throws Exception {
+      performCoreManagement(new ManagementCallback<ClientMessage>() {
+         @Override
+         public void setUpInvocation(ClientMessage message) throws Exception {
+            ManagementHelper.putOperationInvocation(message, "broker", "listQueues", filter, 1, maxRows);
+         }
+
+         @Override
+         public void requestSuccessful(ClientMessage reply) throws Exception {
+            final String result = (String) ManagementHelper.getResult(reply, String.class);
+            printStats(result);
+         }
+
+         @Override
+         public void requestFailed(ClientMessage reply) throws Exception {
+            String errMsg = (String) ManagementHelper.getResult(reply, String.class);
+            context.err.println("Failed to get Stats for Queues. Reason: " + errMsg);
+         }
+      });
+   }
+
+   private void printStats(String result) {
+      printHeadings();
+
+      //should not happen but...
+      if (result == null) {
+         if (verbose) {
+            context.out.println("printStats(): got NULL result string.");
+         }
+      }
+
+      JsonObject queuesAsJsonObject = JsonUtil.readJsonObject(result);
+      JsonArray array = (JsonArray) queuesAsJsonObject.get("data");
+
+      for (int i = 0; i < array.size(); i++) {
+         printQueueStats(array.getJsonObject(i));
+      }
+   }
+
+   private void printHeadings() {
+
+      StringBuilder stringBuilder = new StringBuilder(106).append('|').append(paddingString(new StringBuilder("QUEUE NAME "), 25)).append('|').append(paddingString(new StringBuilder("ADDRESS NAME "), 25)).append('|').append(paddingString(new StringBuilder("CONSUMERS "), 10)).append('|').append(paddingString(new StringBuilder("MESSAGES "), 9)).append('|').append(paddingString(new StringBuilder("ADDED "), 9)).append('|').append(paddingString(new StringBuilder("DELIVERING "), 11)).append('|').append(paddingString(new StringBuilder("ACKED "), 9)).append('|');
```

Comment:
Should headers not be the fieldNames? This then ties in with my comment about more generic filter support, to support ability to filter by any fieldName

Reply:
@michaelandrepearce  Thanks Mike. I will change it to "--field queueName --operation EQUALS --value Test1"  and filter on columns (the set returned).  I was originally going for very simple little options but that will be more useful. I hope to get to that tomorrow.